pkg/controller/networkpolicy/validate.go

```diff
@@ -309,6 +316,22 @@ func (v *antreaPolicyValidator) validateRuleName(ingress, egress []secv1alpha1.R
 	return isUnique(ingress) && isUnique(egress)
 }
 
+func (a *antreaPolicyValidator) validateAppliedTo(ingress, egress []secv1alpha1.Rule, specAppliedTo []secv1alpha1.NetworkPolicyPeer) (string, bool) {
+	appliedToInSpec, appliedToInRules := specAppliedTo != nil, false
+	for i, rule := range append(ingress, egress...) {
+		if i == 0 {
+			appliedToInRules = rule.AppliedTo != nil
+		}
+		if (rule.AppliedTo != nil) != appliedToInRules {
+			return fmt.Sprintf("appliedTo field does not appear consistently in all rules"), false
+		}
+	}
+	if appliedToInSpec == appliedToInRules {
+		return fmt.Sprintf("appliedTo is set in both spec and rules"), false
+	}
```

Comment:
I would avoid building a new slice with append, even if it's more verbose:
```go
appliedToInSpec := specAppliedTo != nil
countAppliedToInRules := func(rules []secv1alpha1.Rule) int {
    num := 0
    for _, rule := range rules {
        if rule.AppliedTo != nil {
            num++
        }
    }
    return num
}
numAppliedToInRules := countAppliedToInRules(ingress) + countAppliedToInRules(egress)
if appliedToInSpec && (numAppliedToInRules > 0) {
    return "appliedTo is set in both spec and rules", false
}
if numAppliedToInRules > 0 && (numAppliedToInRules != len(ingress) + len(egress)) {
    return "appliedTo field should either be set in all rules or in none of them", false
}
```

I think we had a similar discussion for a previous PR, but maybe that's just premature optimization

note; you don't seem to need `fmt.Sprintf` since you just want to return a plain string

Reply:
Thanks for the suggestion. Updated.